<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Anna V. Protasio" />

<meta name="date" content="2019-03-10" />

<title>SEACR peaks</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="02_SEACR_peaks.html">SEACR Peaks</a>
</li>
<li>
  <a href="03_Ruvseq_diffEnrich.html">RUVseq and diff enrichment</a>
</li>
<li>
  <a href="04_overlaps.html">Overlaps</a>
</li>
<li>
  <a href="05_geneExpression.html">Gene Expression</a>
</li>
<li>
  <a href="04b_overlaps_TASOR.html">Overlaps TASOR peaks</a>
</li>
<li>
  <a href="05b_geneExpression_TASOR.html">Gene Expression TASOR peaks</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">SEACR peaks</h1>
<h4 class="author">Anna V. Protasio</h4>
<h4 class="date">03/10/2019</h4>

</div>


<div id="load-peak-data-and-exclude-unwanted-regions" class="section level2">
<h2>Load peak data and exclude unwanted regions</h2>
<p>The peak files have been produced using SEACR <a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>. Brifley, after peak call, peaks of less than 1000nt were excluded from the final set in the understanding that &lt; 1kb peaks are <strong>not</strong> bona fide H<sub>3</sub>K<sub>9</sub>me<sup>3</sup> marks.</p>
<p>Use telomere and centromere coordinates to remove these from the peak sets found by SEACR.</p>
<p>To obtain the coords for telomeres and centromeres I used <a href="https://genome.ucsc.edu" class="uri">https://genome.ucsc.edu</a> server. <em>This information can be found in the “gap” database table. Use the Table Browser (<a href="https://genome.ucsc.edu/cgi-bin/hgTables" class="uri">https://genome.ucsc.edu/cgi-bin/hgTables</a>) to extract it. To do this, select your assembly, in ‘group’ select ‘All Tables’, in ‘tables’ select ‘gap’, in ‘filter’ click ‘Create’ button. Set the ‘type’ field to ‘centromere telomere’ (separated by a space).</em> By doing this, I only managed to obtain the telomeres coordinates. Save file as BED format.</p>
<p>For centromeres, I use the coordinates provided in this pakcage <a href="https://www.rdocumentation.org/packages/GWASTools/versions/1.18.0" class="uri">https://www.rdocumentation.org/packages/GWASTools/versions/1.18.0</a>; function <code>data(centromeres.hg38)</code></p>
<p>Write the new filtered peaks files to a new directory.</p>
<p>Modify sample sheet. New sample sheet looks like:</p>
<table class="table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
SampleID
</th>
<th style="text-align:left;">
Tissue
</th>
<th style="text-align:left;">
Condition
</th>
<th style="text-align:right;">
Replicate
</th>
<th style="text-align:left;">
Peaks
</th>
<th style="text-align:left;">
PeakCaller
</th>
<th style="text-align:left;">
Treatment
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
D3-H3K9me3-rep1
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
D3
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/D3-H3K9me3-rep1.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
ctrl
</td>
</tr>
<tr>
<td style="text-align:left;">
D3-H3K9me3-rep2
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
D3
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/D3-H3K9me3-rep2.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
ctrl
</td>
</tr>
<tr>
<td style="text-align:left;">
TKO-H3K9me3-rep1
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
TKO
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/TKO-H3K9me3-rep1.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
exp
</td>
</tr>
<tr>
<td style="text-align:left;">
TKO-H3K9me3-rep2
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
TKO
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/TKO-H3K9me3-rep2.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
exp
</td>
</tr>
<tr>
<td style="text-align:left;">
WTT-H3K9me3-rep1
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
WTT
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/WTT-H3K9me3-rep1.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
ctrl
</td>
</tr>
<tr>
<td style="text-align:left;">
WTT-H3K9me3-rep2
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
WTT
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/WTT-H3K9me3-rep2.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
ctrl
</td>
</tr>
<tr>
<td style="text-align:left;">
mutT-H3K9me3-rep1
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
mutT
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/mutT-H3K9me3-rep1.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
exp
</td>
</tr>
<tr>
<td style="text-align:left;">
mutT-H3K9me3-rep2
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
mutT
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/mutT-H3K9me3-rep2.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
exp
</td>
</tr>
<tr>
<td style="text-align:left;">
D3-H3K9me3-rep3
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
D3
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/D3-H3K9me3-rep3.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
ctrl
</td>
</tr>
<tr>
<td style="text-align:left;">
D3-H3K9me3-rep4
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
D3
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/D3-H3K9me3-rep4.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
ctrl
</td>
</tr>
<tr>
<td style="text-align:left;">
TKO-H3K9me3-rep3
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
TKO
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/TKO-H3K9me3-rep3.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
exp
</td>
</tr>
<tr>
<td style="text-align:left;">
TKO-H3K9me3-rep4
</td>
<td style="text-align:left;">
HeLa
</td>
<td style="text-align:left;">
TKO
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
data/Peaks-seacr-filter/TKO-H3K9me3-rep4.stringent.1000long.bed
</td>
<td style="text-align:left;">
bed
</td>
<td style="text-align:left;">
exp
</td>
</tr>
</tbody>
</table>
<p>Use clean set of SEACR peaks to find a consensus set of peaks based on the ctrl samples (D3 and WTT)</p>
<p>The following plots shows the number of peaks for the control samples (D3 plus WTT) and the overlap rate. We make use of some of the practical functionalities of the <code>R package DiffBind</code> to find the consensus peaks based on their coordinates overlap.</p>
<p><img src="02_SEACR_peaks_files/figure-html/plot_olap_ctrl-1.png" width="672" /></p>
<p><img src="02_SEACR_peaks_files/figure-html/plot_olap_d3-1.png" width="672" /></p>
<p>Ideally, the number of peaks should plato as the number of overlaps increases. This is not the case for these samples probably because there is much variation among samples. Therefore, we proceed with a permissive set of peak as we will only focus on those that that are differentially enriched/bound.</p>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] kableExtra_1.1.0            DEScan_0.99.0              
##  [3] GWASTools_1.30.1            gdata_2.18.0               
##  [5] DESeq2_1.24.0               DiffBind_2.12.0            
##  [7] SummarizedExperiment_1.14.1 DelayedArray_0.10.0        
##  [9] BiocParallel_1.18.1         matrixStats_0.55.0         
## [11] Biobase_2.44.0              GenomicRanges_1.36.1       
## [13] GenomeInfoDb_1.20.0         IRanges_2.18.3             
## [15] S4Vectors_0.22.1            BiocGenerics_0.30.0        
## [17] knitr_1.25                 
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.1.5          GOstats_2.50.0          
##   [3] Hmisc_4.2-0              lazyeval_0.2.2          
##   [5] GSEABase_1.46.0          splines_3.6.1           
##   [7] ggplot2_3.2.1            amap_0.8-17             
##   [9] digest_0.6.21            htmltools_0.4.0         
##  [11] GO.db_3.8.2              magrittr_1.5            
##  [13] checkmate_1.9.4          memoise_1.1.0           
##  [15] BSgenome_1.52.0          base64url_1.4           
##  [17] cluster_2.1.0            limma_3.40.6            
##  [19] readr_1.3.1              Biostrings_2.52.0       
##  [21] annotate_1.62.0          systemPipeR_1.18.2      
##  [23] sandwich_2.5-1           prettyunits_1.0.2       
##  [25] colorspace_1.4-1         rvest_0.3.4             
##  [27] blob_1.2.0               rappdirs_0.3.1          
##  [29] ggrepel_0.8.1            pan_1.6                 
##  [31] xfun_0.10                dplyr_0.8.3             
##  [33] crayon_1.3.4             RCurl_1.95-4.12         
##  [35] graph_1.62.0             lme4_1.1-21             
##  [37] genefilter_1.66.0        zeallot_0.1.0           
##  [39] brew_1.0-6               survival_2.44-1.1       
##  [41] VariantAnnotation_1.30.1 zoo_1.8-6               
##  [43] glue_1.3.1               gtable_0.3.0            
##  [45] zlibbioc_1.30.0          XVector_0.24.0          
##  [47] webshot_0.5.1            Rsubread_1.34.7         
##  [49] MatrixModels_0.4-1       Rgraphviz_2.28.0        
##  [51] SparseM_1.77             jomo_2.6-9              
##  [53] scales_1.0.0             pheatmap_1.0.12         
##  [55] DBI_1.0.0                edgeR_3.26.8            
##  [57] Rcpp_1.0.2               viridisLite_0.3.0       
##  [59] xtable_1.8-4             progress_1.2.2          
##  [61] htmlTable_1.13.2         quantsmooth_1.50.0      
##  [63] foreign_0.8-72           bit_1.1-14              
##  [65] Formula_1.2-3            AnnotationForge_1.26.0  
##  [67] htmlwidgets_1.5.1        httr_1.4.1              
##  [69] logistf_1.23             gplots_3.0.1.1          
##  [71] RColorBrewer_1.1-2       acepack_1.4.1           
##  [73] mice_3.6.0               pkgconfig_2.0.3         
##  [75] XML_3.98-1.20            nnet_7.3-12             
##  [77] locfit_1.5-9.1           DNAcopy_1.58.0          
##  [79] tidyselect_0.2.5         rlang_0.4.0             
##  [81] AnnotationDbi_1.46.1     munsell_0.5.0           
##  [83] tools_3.6.1              generics_0.0.2          
##  [85] RSQLite_2.1.2            broom_0.5.2             
##  [87] evaluate_0.14            stringr_1.4.0           
##  [89] yaml_2.2.0               bit64_0.9-7             
##  [91] gdsfmt_1.20.0            caTools_1.17.1.2        
##  [93] purrr_0.3.2              mitml_0.3-7             
##  [95] RBGL_1.60.0              nlme_3.1-141            
##  [97] quantreg_5.51            GWASExactHW_1.01        
##  [99] xml2_1.2.2               biomaRt_2.40.5          
## [101] compiler_3.6.1           rstudioapi_0.10         
## [103] tibble_2.1.3             geneplotter_1.62.0      
## [105] stringi_1.4.3            highr_0.8               
## [107] GenomicFeatures_1.36.4   lattice_0.20-38         
## [109] Matrix_1.2-17            nloptr_1.2.1            
## [111] vctrs_0.2.0              lifecycle_0.1.0         
## [113] pillar_1.4.2             lmtest_0.9-37           
## [115] data.table_1.12.2        bitops_1.0-6            
## [117] rtracklayer_1.44.4       R6_2.4.0                
## [119] latticeExtra_0.6-28      hwriter_1.3.2           
## [121] ShortRead_1.42.0         KernSmooth_2.23-15      
## [123] gridExtra_2.3            boot_1.3-23             
## [125] MASS_7.3-51.4            gtools_3.8.1            
## [127] assertthat_0.2.1         Category_2.50.0         
## [129] rjson_0.2.20             withr_2.1.2             
## [131] GenomicAlignments_1.20.1 batchtools_0.9.11       
## [133] Rsamtools_2.0.2          GenomeInfoDbData_1.2.1  
## [135] mgcv_1.8-29              hms_0.5.1               
## [137] grid_3.6.1               rpart_4.1-15            
## [139] minqa_1.2.4              tidyr_1.0.0             
## [141] rmarkdown_1.16           base64enc_0.1-3</code></pre>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p><a href="https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-019-0287-4" class="uri">https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-019-0287-4</a><a href="#fnref1" class="footnote-back">↩</a></p></li>
</ol>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
