<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Anna V. Protasio" />


<title>Count reads / diff binding</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="02_SEACR_peaks.html">SEACR Peaks</a>
</li>
<li>
  <a href="03_Ruvseq_diffEnrich.html">RUVseq and diff enrichment</a>
</li>
<li>
  <a href="04_overlaps.html">Overlaps</a>
</li>
<li>
  <a href="05_geneExpression.html">Gene Expression</a>
</li>
<li>
  <a href="04b_overlaps_TASOR.html">Overlaps TASOR peaks</a>
</li>
<li>
  <a href="05b_geneExpression_TASOR.html">Gene Expression TASOR peaks</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Count reads / diff binding</h1>
<h4 class="author">Anna V. Protasio</h4>
<h4 class="date">28/10/2019</h4>

</div>


<p>Count reads on consensus peaks using the function <code>countBamInGRanges</code> from package <code>exomeCopy</code>. The set of peaks to use is defined by D3 and WTT samples and is loaded from an RData object.</p>
<p>The total number of peaks considered is 41215</p>
<p>The boxplots of relative log expression (RLE = log-ratio of read count to median read count across sample) and plots of principal components (PC) reveal a clear need for between-sample normalization.</p>
<p><img src="03_Ruvseq_diffEnrich_files/figure-html/rawPlot-1.png" width="336" /><img src="03_Ruvseq_diffEnrich_files/figure-html/rawPlot-2.png" width="336" /></p>
<p>The parameter <strong>k</strong> dictates the number of factors of unwanted to variation to remove, in this case we use 3. We can see in the PCA plot that after RUVs normalization the first 2 principal components seperate the four groups indicating that the treatment is NOW the major source of variation.</p>
<p><img src="03_Ruvseq_diffEnrich_files/figure-html/ruvplot-1.png" width="336" /><img src="03_Ruvseq_diffEnrich_files/figure-html/ruvplot-2.png" width="336" /></p>
<div id="testing-for-differential-enrichment-of-regions" class="section level1">
<h1>Testing for differential enrichment of regions</h1>
<p>Now, we are ready to look for differentially enriched regions, using the negative binomial quasi-likelihood GLM approach implemented in edgeR (see the edgeR package vignette for details). This is done by considering a design matrix that includes both the covariates of interest (here, the treatment status) and the factors of unwanted variation.</p>
<div id="tko-vs-d3" class="section level2">
<h2>TKO vs D3</h2>
<p>With a cut-off FDR &lt; 0.05, we find the following numbers of peaks with reduced/increased binding in TKO vs D3.</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
log2FC±0
</th>
<th style="text-align:right;">
log2FC±1
</th>
<th style="text-align:right;">
log2FC±2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
reduced
</td>
<td style="text-align:right;">
943
</td>
<td style="text-align:right;">
393
</td>
<td style="text-align:right;">
76
</td>
</tr>
<tr>
<td style="text-align:left;">
increased
</td>
<td style="text-align:right;">
983
</td>
<td style="text-align:right;">
117
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<p>Simple MA plot.</p>
<p><img src="03_Ruvseq_diffEnrich_files/figure-html/plot1-1.png" width="672" /></p>
</div>
<div id="mutt-vd-d3" class="section level2">
<h2>mutT vd D3</h2>
<p>With a cut-off FDR &lt; 0.05, we find the following numbers of peaks with reduced/increased binding in mutT vs D3.</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
log2FC±0
</th>
<th style="text-align:right;">
log2FC±1
</th>
<th style="text-align:right;">
log2FC±2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
reduced
</td>
<td style="text-align:right;">
698
</td>
<td style="text-align:right;">
415
</td>
<td style="text-align:right;">
107
</td>
</tr>
<tr>
<td style="text-align:left;">
increased
</td>
<td style="text-align:right;">
355
</td>
<td style="text-align:right;">
74
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<p><img src="03_Ruvseq_diffEnrich_files/figure-html/plot2-1.png" width="672" /></p>
</div>
<div id="mutt-vs-tko" class="section level2">
<h2>mutT vs TKO</h2>
<p>With a cut-off FDR &lt; 0.05, we find the following numbers of peaks with reduced/increased binding in mutT vs TKO. (Reduced means lost in mutT with respect of TKO). There are no peaks with increased binding in mutT with respect to TKO.</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
log2FC±0
</th>
<th style="text-align:right;">
log2FC±1
</th>
<th style="text-align:right;">
log2FC±2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
reduced
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
increased
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
</div>
<div id="mutt-vs-wtt" class="section level2">
<h2>mutT vs WTT</h2>
<p>With a cut-off FDR &lt; 0.05, we find the following numbers of peaks with reduced/increased binding in mut vs wtt.</p>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
log2FC±0
</th>
<th style="text-align:right;">
log2FC±1
</th>
<th style="text-align:right;">
log2FC±2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
reduced
</td>
<td style="text-align:right;">
373
</td>
<td style="text-align:right;">
238
</td>
<td style="text-align:right;">
46
</td>
</tr>
<tr>
<td style="text-align:left;">
increased
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<p>Simple MA plot.</p>
<p><img src="03_Ruvseq_diffEnrich_files/figure-html/plot4-1.png" width="672" /></p>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] exomeCopy_1.30.0            ggplot2_3.2.1              
##  [3] tidyr_1.0.0                 kableExtra_1.1.0           
##  [5] statmod_1.4.32              RUVSeq_1.18.0              
##  [7] edgeR_3.26.8                limma_3.40.6               
##  [9] EDASeq_2.18.0               ShortRead_1.42.0           
## [11] GenomicAlignments_1.20.1    SummarizedExperiment_1.14.1
## [13] DelayedArray_0.10.0         matrixStats_0.55.0         
## [15] Rsamtools_2.0.2             Biostrings_2.52.0          
## [17] XVector_0.24.0              BiocParallel_1.18.1        
## [19] Biobase_2.44.0              DEScan2_1.4.0              
## [21] GenomicRanges_1.36.1        GenomeInfoDb_1.20.0        
## [23] IRanges_2.18.3              S4Vectors_0.22.1           
## [25] BiocGenerics_0.30.0         RColorBrewer_1.1-2         
## [27] knitr_1.25                 
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_1.4-1       seqinr_3.6-1           hwriter_1.3.2         
##   [4] futile.logger_1.4.3    rstudioapi_0.10        ChIPpeakAnno_3.18.2   
##   [7] bit64_0.9-7            AnnotationDbi_1.46.1   xml2_1.2.2            
##  [10] splines_3.6.1          R.methodsS3_1.7.1      DESeq_1.36.0          
##  [13] geneplotter_1.62.0     zeallot_0.1.0          ade4_1.7-13           
##  [16] annotate_1.62.0        GO.db_3.8.2            R.oo_1.22.0           
##  [19] graph_1.62.0           BiocManager_1.30.7     readr_1.3.1           
##  [22] compiler_3.6.1         httr_1.4.1             backports_1.1.5       
##  [25] assertthat_0.2.1       Matrix_1.2-17          lazyeval_0.2.2        
##  [28] formatR_1.7            htmltools_0.4.0        prettyunits_1.0.2     
##  [31] tools_3.6.1            gtable_0.3.0           glue_1.3.1            
##  [34] GenomeInfoDbData_1.2.1 dplyr_0.8.3            Rcpp_1.0.2            
##  [37] vctrs_0.2.0            multtest_2.40.0        rtracklayer_1.44.4    
##  [40] xfun_0.10              stringr_1.4.0          rvest_0.3.4           
##  [43] lifecycle_0.1.0        ensembldb_2.8.1        XML_3.98-1.20         
##  [46] idr_1.2                zlibbioc_1.30.0        MASS_7.3-51.4         
##  [49] scales_1.0.0           aroma.light_3.14.0     BSgenome_1.52.0       
##  [52] hms_0.5.1              ProtGenerics_1.16.0    RBGL_1.60.0           
##  [55] AnnotationFilter_1.8.0 lambda.r_1.2.4         yaml_2.2.0            
##  [58] curl_4.2               memoise_1.1.0          biomaRt_2.40.5        
##  [61] latticeExtra_0.6-28    stringi_1.4.3          RSQLite_2.1.2         
##  [64] highr_0.8              genefilter_1.66.0      GenomicFeatures_1.36.4
##  [67] rlang_0.4.0            pkgconfig_2.0.3        bitops_1.0-6          
##  [70] evaluate_0.14          lattice_0.20-38        purrr_0.3.2           
##  [73] labeling_0.3           tidyselect_0.2.5       bit_1.1-14            
##  [76] plyr_1.8.4             magrittr_1.5           R6_2.4.0              
##  [79] DBI_1.0.0              withr_2.1.2            pillar_1.4.2          
##  [82] survival_2.44-1.1      RCurl_1.95-4.12        tibble_2.1.3          
##  [85] crayon_1.3.4           futile.options_1.0.1   rmarkdown_1.16        
##  [88] progress_1.2.2         locfit_1.5-9.1         grid_3.6.1            
##  [91] data.table_1.12.2      blob_1.2.0             digest_0.6.21         
##  [94] webshot_0.5.1          xtable_1.8-4           VennDiagram_1.6.20    
##  [97] regioneR_1.16.5        R.utils_2.9.0          munsell_0.5.0         
## [100] viridisLite_0.3.0</code></pre>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
